src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -0,0 +1,122 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import cli_command
+from azure.cli.core._util import CLIError
+
+from azure.cli.command_modules.acr.containerregistry.models import (
+    RegistryParameters,
+    RegistryProperties,
+    StorageAccountProperties
+)
+
+from ._factory import (
+    get_registry_service_client,
+    get_storage_end_point_suffix
+)
+
+from ._arm_utils import (
+    arm_get_registries_in_subscription,
+    arm_get_registries_in_resource_group,
+    arm_get_registry_by_name,
+    arm_deploy_template
+)
+
+from ._utils import (
+    get_registry_by_name,
+    get_resource_group_by_registry
+)
+
+from ._format import output_format
+
+def acr_list(resource_group=None):
+    '''Returns the list of container registries.
+    :param str resource_group: The name of resource group
+    '''
+    if resource_group:
+        return arm_get_registries_in_resource_group(resource_group)
+    else:
+        return arm_get_registries_in_subscription()
+
+def acr_create(resource_group, registry_name, location, #pylint: disable=too-many-arguments
+               storage_account_name=None, storage_account_key=None):
```

Comment:
>storage_account_name=None, storage_account_key=None [](start = 15, length = 51)

You should also be able to specify a storage account connection string here I would think (maybe a SAS token?)